pkg/harvester/components/settings/rancher-cluster.vue

```diff
@@ -0,0 +1,253 @@
+        this.$nextTick(() => {
+          this.update();
+        });
+      }
+    },
+
+    async createOrUpdateRancherKubeConfigSecret(buttonCb) {
+      this.errors = [];
+      this.isCreatingSecret = true;
+
+      if (!this.parseDefaultValue.kubeConfig) {
+        this.errors.push(this.t('validation.required', { key: this.t('harvester.setting.rancherCluster.kubeConfig') }, true));
+        buttonCb(false);
+        this.isCreatingSecret = false;
+
+        return;
+      }
+
+      try {
+        const inStore = this.$store.getters['currentProduct'].inStore;
+
+        let secret;
+        const isUpdate = !!this.existingSecret;
+
+        if (this.existingSecret) {
+          // Update existing secret
+          secret = this.existingSecret;
+          secret.setData('kubeConfig', this.parseDefaultValue.kubeConfig);
+        } else {
+          // Create new secret
+          secret = await this.$store.dispatch(`${ inStore }/create`, {
+            apiVersion: 'v1',
+            kind:       'Secret',
+            metadata:   {
+              name:      'rancher-cluster-config',
+              namespace: 'harvester-system'
+            },
+            type: 'secret',
+            data: { kubeConfig: btoa(this.parseDefaultValue.kubeConfig) }
+          });
+        }
+
+        await secret.save();
+
+        // Refresh the secret list and update the existing secret reference
+        await this.checkExistingSecret();
+
+        const message = isUpdate ? this.t('harvester.setting.rancherCluster.secretUpdated') : this.t('harvester.setting.rancherCluster.secretCreated');
+
+        this.$store.dispatch('growl/success', { message }, { root: true });
+
+        buttonCb(true);
+      } catch (err) {
+        this.errors = exceptionToErrorsArray(err);
+        buttonCb(false);
+      } finally {
+        this.isCreatingSecret = false;
+      }
+    }
+  },
+
+  watch: {
+    value: {
+      handler(neu) {
+        const parsed = JSON.parse(neu.value || neu.default || '{}');
+
+        // Keep kubeConfig field for UI, but only save removeUpstreamClusterWhenNamespaceIsDeleted
+        this['parseDefaultValue'] = {
+          kubeConfig:                                  this.parseDefaultValue?.kubeConfig || '',
+          removeUpstreamClusterWhenNamespaceIsDeleted: parsed.removeUpstreamClusterWhenNamespaceIsDeleted || false
+        };
+      },
+      deep: true
+    }
+  }
+};
+</script>
+
+<template>
+  <div>
+    <Banner color="info">
+      {{ t('harvester.setting.rancherCluster.description') }}
+    </Banner>
+
+    <div class="row mt-20">
+      <div class="col span-12">
+        <FileSelector
+          class="btn btn-sm bg-primary mb-10"
+          :label="t('generic.readFromFile')"
+          @selected="onKeySelected"
+        />
+
+        <div>
+          <TextAreaAutoGrow
+            v-model:value="parseDefaultValue.kubeConfig"
+            :min-height="254"
+            @update:value="update"
+          />
```

Comment:
Are users allowed to modify the KubeConfig? I don’t see it included in the request body.